core/metrics.go

```diff
@@ -0,0 +1,152 @@
+/*
+   Copyright (c) 2015 Andrey Sibiryov <me@kobology.ru>
+   Copyright (c) 2015 Other contributors as noted in the AUTHORS file.
+
+   This file is part of GORB - Go Routing and Balancing.
+
+   GORB is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published by
+   the Free Software Foundation; either version 3 of the License, or
+   (at your option) any later version.
+
+   GORB is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package core
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	promNamespace = "gorb"
+	defaultSleep  = time.Second * 5
+)
+
+var (
+	lbHealth = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "lbs_health",
+		Help:      "load balancers health percentage",
+	}, []string{"lb_name", "host", "port", "protocol", "persistent", "method"})
+	backendNumber = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "lbs_backends_number",
+		Help:      "load balancers backends informations",
+	}, []string{"lb_name", "host", "port", "method"})
+	backendWeight = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "lbs_backends_weight",
+		Help:      "load balancers backends weight",
+	}, []string{"lb_name", "backend_name", "host", "port", "method"})
+	backendStatus = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "lbs_backends_status",
+		Help:      "load balancers backends status",
+	}, []string{"lb_name", "backend_name", "host", "port", "method"})
+	backendHealth = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: promNamespace,
+		Name:      "lbs_backends_health",
+		Help:      "load balancers backends health",
+	}, []string{"lb_name", "backend_name", "host", "port", "method"})
+	lbChannelMap      = make(map[string]chan int)
+	backendChannelMap = make(map[string]chan int)
+)
+
+// Prometheus time series
+func init() {
+	prometheus.MustRegister(lbHealth)
+	prometheus.MustRegister(backendNumber)
+	prometheus.MustRegister(backendWeight)
+	prometheus.MustRegister(backendStatus)
+	prometheus.MustRegister(backendHealth)
+}
+
+func StartLBMetric(ctx *Context, vsID string) {
+	lbChannelMap[vsID] = make(chan int)
```

Comment:
Instead of using integers and than throwing away the value passed on the channel, just use empty struct{} like explained here: https://blog.golang.org/pipelines. 

Ok, make sense. I'll fix this.